### ISSUE TITLE
feature: Add option to specify initial number of lines in pyodide fence editors

### ISSUE THREAD

**Issue by pawamoy** [pawamoy/markdown-exec#41]
### Is your feature request related to a problem? Please describe.

If I want to add an editor with just three lines, I'd like to spare some vertical space. Currently it always has the same number of lines.

### Describe the solution you'd like

Allow configuring number of lines in pyodide editors.

### Describe alternatives you've considered
/

### Additional context
/

**Comment by michaelweinold** [comment]
I, too, like vertical space 💙

I have put [a 25$ bounty on this issue over at gitpay.io](https://gitpay.me/#/task/1079/feature-add-option-to-specify-initial-number-of-lines-in-pyodide-fence-editors).

Perhaps three options could be enabled?

1. Default vertical size (9-lines?)
2. Custom vertical size, user-defined
3. As much vertical space as lines of code in the fence

**Comment by waimea-cpy** [comment]
Agree... Came here to ask if this was possible via any sort of config. 

For me the issue is that larger blocks of code only show a few lines and have to be scrolled, so the suggested option 3 from @michaelweinold would be ideal.

**Comment by pawamoy** [comment]
Thanks for setting a bounty again @michaelweinold!

Just to be transparent: features marked with the Insiders label are generally released to recurring sponsors first, as Insiders features. This sponsorware strategy is explained here: https://pawamoy.github.io/insiders/. Since this might not have been clear, I'm willing to release this feature to everyone immediately instead of releasing it to sponsors first. But if you plan on setting up bounties on other features marked with the Insiders label, please know I'll release them to sponsors first 🙂 The reason is that without Insiders features, I don't get recurring sponsors, and without recurring sponsors, my ability to work on all these open-source projects is impeded. I hope that makes sense 😄 Happy to get your thoughts on this and improve the process any way I can.

---

The initial idea was point 2, custom vertical size, user-defined, per code block. Point 1 is a good idea, but will need some more general thinking about how to provide default values (which are requested more and more). I'll open a new issue for that.

Point 3 is a good idea too, though needs some clarifications:

- is the height fixed (based on initial code), or does it update when new lines are added?
- do we set a maximum? I think we should. A 100+ line high editor doesn't sound like a good UX 🤔 
- similarly, should it shrink back to a minimum height when removing lines?

**Comment by michaelweinold** [comment]
The options look good - I would personally be very happy with those already.

**Comment by amit-s19** [comment]
@michaelweinold I've created a PR for this, can you please check if this looks good? #85

**Comment by waimea-cpy** [comment]
Thanks for the clarification re. the 'insiders' tag, and I hope you don't mind my chiming in!

When I was looking to see if I could implement this myself by modifying the settings for the Ace Editor instances, I came across this demo which shows off various sizing features, plus one that auto-resizes as the code is edited: https://ace.c9.io/demo/autoresize.html Thought it might be helpful...

**Comment by michaelweinold** [comment]
> @michaelweinold I've created a PR for this, can you please check if this looks good? https://github.com/pawamoy/markdown-exec/pull/85

Thank you for opening the PR. I will leave the assessment (and obviously merging) of this feature branch to the package maintainer @pawamoy. He is, I believe, also working on this feature and may therefore opt to go with his solution instead.

**Comment by amit-s19** [comment]
@pawamoy don't call me greedy but I came here from Gitpay 😅

**Comment by pawamoy** [comment]
@amit-s19 I thought it was a possibility 😄 And no worries, that's completely fine 🙂 I'm happy if you get the bounty!

**Comment by waimea-cpy** [comment]
I was looking to see if there were any open issues related to pyodide before I posted a feature request.... So I used the issue search, and this came up.

**Comment by suraj-mandal** [comment]
I can see that the bounty is still open on githpay. Is the PR merged?

**Comment by Zedoman** [comment]
I have added a PR to solve this. @michaelweinold

**Comment by jayantpranjal0** [comment]
@pawamoy can I work on this issue?

**Comment by pawamoy** [comment]
@michaelweinold the feature was implemented and released 🙂 If you want to let us know how it goes.

````
```pyodide height="1-20"
print("initial code")
```
````
